fml/platform/win/file_win.cc

```diff
@@ -20,7 +25,46 @@
 
 namespace fml {
 
+#ifdef WINUWP
+struct DirCacheEntry {
+  std::wstring filename;
+  FILE_ID_128 id;
+};
+std::mutex file_map_mutex;
+static std::map<HANDLE, DirCacheEntry> file_map;
```

Comment:
Nit: `gFileMap`

Reply:
Thx for the understanding.  I've got most of the rework done apart from splitting the UWP/win32 aspects into different files.  Will clean up and push for your consideration.

Reply:
Change pushed.  I couldn't figure out how to run the tests in `flutter\fml\file_unittest.cc`.  Once I know how to do that will add some tests for the caching (I've debugged through the code paths and it's working as expected).  LMK any remaining changes you'd like to see to land and ultimately to be clean.

Reply:
The unit-tests for each subsystem are discrete gtest binaries that you can invoke manually. For FML, this would be `./out/<variant>/fml_unittests.exe`. This should run all FML unit-tests. If you run `./flutter/testesting/run_tests.py` with no arguments, it will run all tests (with flags to select specific harnesses).

Reply:
Roger, thx will add a test.

Reply:
I've run through the existing fml file tests for the `winuwp` target and verified that the new path is completely covered by existing tests hence resolving this conversation